.github/workflows/release.yml

```
@@ -54,11 +60,18 @@ jobs:
         uses: rlespinasse/github-slug-action@v4
 
       - name: Set up Go
+        if: ${{ matrix.goos != 'windows' || matrix.os-variant != '7' }}
         uses: actions/setup-go@v5
         with:
           go-version: '^1.22.1'
 
-      - run: echo "basename=sni-${{env.GITHUB_REF_SLUG}}-${{matrix.goos}}-${{matrix.goarch}}${{matrix.alt}}" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+      - name: Set up Go
+        if: ${{ matrix.goos == 'windows' && matrix.os-variant == '7' }}
+        uses: actions/setup-go@v5
+        with:
+          go-version: '^1.20.14'
```

This should be the correct syntax for it
```suggestion
          go-version: '~1.20.14'
```